instrumentation/log/filters.go

```diff
@@ -97,5 +110,30 @@ func (f *excludeField) Allows(level string, message string, fields []*Field) boo
 	return true
 }
 
+type matchField struct {
```

Comment:
We have `includeFieldWithKey`. Can you think of a nice way to generalize a type to handle both this case and the other case?